lib/ex_aws/cloudwatch.ex

```diff
@@ -1,42 +1,603 @@
+    alarm_description: binary,
+    dimensions: [dimension, ...],
+    evaluate_low_sample_count_percentile: binary,
+    extended_statistic: binary,
+    insufficient_data_actions: [binary, ...],
+    ok_actions: [binary, ...],
+    treat_missing_data: binary,
+    unit: binary
+  ]
+  @spec put_metric_alarm(alarm_name :: binary, comparison_operator :: binary, evaluation_periods :: integer, metric_name :: binary, namespace :: binary, period :: integer, threshold :: float, statistic :: binary) :: ExAws.Operation.Query.t()
+  @spec put_metric_alarm(alarm_name :: binary, comparison_operator :: binary, evaluation_periods :: integer, metric_name :: binary, namespace :: binary, period :: integer, threshold :: float, statistic :: binary, opts :: put_metric_alarm_opts) :: ExAws.Operation.Query.t()
+  def put_metric_alarm(alarm_name, comparison_operator, evaluation_periods, metric_name, namespace, period, threshold, statistic, opts \\ []) do
+    [
+      {:alarm_name, alarm_name},
+      {:comparison_operator, comparison_operator},
+      {:evaluation_periods, evaluation_periods},
+      {:metric_name, metric_name},
+      {:namespace, namespace},
+      {:period, period},
+      {:threshold, threshold},
+      {:statistic, statistic} | opts
+    ]
+    |> build_request(:put_metric_alarm)
+  end
+
+  @doc """
+  Publishes metric data points to Amazon CloudWatch.
+
+  CloudWatch associates the data points with the specified metric.
+  If the specified metric does not exist, CloudWatch creates the metric.
+  When CloudWatch creates a metric, it can take up to fifteen minutes for
+  the metric to appear in calls to ListMetrics .
+
+  Each PutMetricData request is limited to 40 KB in size for HTTP POST requests.
+
+  Although the Value parameter accepts numbers of type Double, CloudWatch rejects
+  values that are either too small or too large. Values must be in the range
+  of 8.515920e-109 to 1.174271e+108 (Base 10) or 2e-360 to 2e360 (Base 2).
+  In addition, special values (for example, NaN, +Infinity, -Infinity) are
+  not supported.
+
+  You can use up to 10 dimensions per metric to further clarify what data the
+  metric collects. For more information about specifying dimensions, see
+  Publishing Metrics in the Amazon CloudWatch User Guide.
+
+  Data points with time stamps from 24 hours ago or longer can take at least
+  48 hours to become available for GetMetricStatistics from the time they are
+  submitted.
+
+  CloudWatch needs raw data points to calculate percentile statistics. If you
+  publish data using a statistic set instead, you can only retrieve percentile
+  statistics for this data if one of the following conditions is true:
+
+  * The SampleCount value of the statistic set is 1
+  * The Min and the Max values of the statistic set are equal
+
+
+  """
+  @spec put_metric_data_input(metric_data :: [metric_datum, ...], namespace :: binary) :: ExAws.Operation.Query.t()
+  def put_metric_data_input(metric_data, namespace) do
```

Comment:
Should this be put_metric_data instead?  https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_PutMetricData.html

Reply:
shoot. you are absolutely correct. this was one method that I haven't used so I didn't notice that it was the wrong name. I've changed it.